### ISSUE TITLE
Ensure a trusted light block stays trusted permanently, even if its peer fails later

### ISSUE THREAD

**Issue by romac** [informalsystems/tendermint-rs#420]
> > @milosevic I had a bit of different issue in mind, which more about how we deal with state of different peers. During the `Supervisor` verification the only instance which updates it's `LightBlockStore` is the primary, while during fork detection the witnesses are using freshly created in memory stores which after that get discarded. Therefore, even though we have stored a `LightBlock` which we consider trusted if at any point in time we find out the primary is faulty we swap it and use the `State` associated with the new primary instance and as soon as we do that we basically go back in time to the state before any verification. For me this behaviours is strange as the history of previously verified `LightBlocks` is lost and we might need to restart the light client just because the primary was faulty.
> 
> @josef-widder  I think once a LightBlock is trusted, it should stay trusted forever, even if the primary fails later. Otherwise, it is not clear what is the meaning of "trusted".

— https://github.com/informalsystems/tendermint-rs/issues/395#issuecomment-653508642

**Comment by xla** [comment]
This should include the API to not return an `Option`.

**Comment by milosevic** [comment]
I agree that trusted light block is permanently trusted, even after primary fails (we don't consider forks in this case). We should rely on this invariant when electing new primary; we can continue operating from the latest trusted state, i.e., we don't need to throw away complete state and reinitialise the light client.
